RCP-042: TLS v1.3 RFC updates

grispin opened the pull request:
Updates to WebApi standards to current industry standards.  The current reference document has security concerns.

darnjo reviewed (changes requested):
I believe the current best practice recommendations are that servers should support 1.2 and greater, and a [quick Google search](https://www.google.com/search?q=minimum+TLS+version&oq=minimum+TLS+version&aqs=chrome..69i57.2796j0j1&sourceid=chrome&ie=UTF-8) seems to support this. 

If changes were going to be made, I would suggest that the text be something to the effect of: "Servers MAY support 1.3 but MUST support at least 1.2."

cc: @psftc

grispin commented:
I'm not saying that we support only TLS v1.3 but we make it the mandatory version for certification so the the clients and servers negotiate the most secure connection possible.  TLSv1.3 has been out as a standard for 4 years now and is included in every major browser and web platform engine.  

The TLS v1.2 list of vulnerabilities is getting longer with a large portion of the ciphers in TLS v1.2 are considered weak due to known issues  [Wikipedia TLS page](https://en.wikipedia.org/wiki/Transport_Layer_Security)  There is significant work that both clients and servers must do to ensure an TLS v1.2 compliant library is actually secure.   Specifying TLSv1.3 addresses those issues in the longer term as they are all included in the updated specification.  

If we want to allow TLSv1.2 for a period of time for backwards compatibility, I can understand that but we should make TLSv1.3 a "MUST" for the next release so the standard moves forward with

grispin commented:
@darnjo I would still like for us to get this into the specification.  If not this release, then the next one.
